pkg/dataplane/client.go

```diff
@@ -35,6 +63,63 @@ func NewClient(aud, cloud string, cred azcore.TokenCredential) (*ManagedIdentity
 	return &ManagedIdentityClient{swaggerClient: swaggerClient}, nil
 }
 
+func (c *ManagedIdentityClient) GetUserAssignedMSI(ctx context.Context, request UserAssignedMSIRequest) (*CredentialsObject, error) {
+	validate := validator.New(validator.WithRequiredStructEnabled())
+	if err := validate.Struct(request); err != nil {
+		return nil, fmt.Errorf("%w: %w", errInvalidRequest, err)
+	}
+
+	ctx = context.WithValue(ctx, identityURLKey, request.IdentityURL)
+	credRequestDef := swagger.CredRequestDefinition{
+		IdentityIDs: []*string{&request.ResourceID},
+	}
+
+	creds, err := c.swaggerClient.Getcreds(ctx, credRequestDef, nil)
```

Comment:
should this be `GetCreds`?

Reply:
No- the generated swagger client names this `Getcreds`: https://github.com/Azure/msi-dataplane/blob/83c74bf44a052b0dcba186ea859b87472be68ad7/internal/swagger/zz_generated_managedidentitydataplaneapi_client.go#L112